documentation/guide-configuration.asciidoc

```diff
@@ -156,47 +156,189 @@ We recommend the following base layout for the hierarchical business configurati
 ----
 
-The master-password can be configured on your target environment via the property `jasypt.encryptor.password`. As system properties given on the command-line are visible in the process list, we recommend to use an `config/application.yml` file only for this purpose (as we recommended to use `application.properties` for regular configs):
-```
+Let's now encrypt the text`Password@1` with secret key`password`  and add it to the `encrypted.properties`:
+[source,properties]
+----
+encrypted.property=ENC(uTSqb9grs1+vUv3iN8lItC0kl65lMG+8)
+----
+
+And let's define a configuration class `ConfigForJasyptStarter` – to specify the `encrypted.properties` file as a PropertySource :
+
+[source,java]
+----
+@Configuration
+@PropertySource("encrypted.properties")
+public class AppConfigForJasyptStarter {
+}
+----
+
+Let's create service bean class to retrieve the values from `encrypted.properties` file.
+The decrypted value can be retrieved using the `@Value` annotation or the getProperty() method of `Environment` class. See below example:
+
+[source,java]
+----
+@Service
+public class PropertyMapperJasyptStarter {
+
+    @Value("${encrypted.property}")
+    private String property;
+
+    public String getProperty() {
+        return property;
+    }
+
+    public String getPasswordUsingEnvironment(Environment environment) {
+        return environment.getProperty("encrypted.property");
+    }
+}
+----
+
+Using the above service class and setting the secret key which we used for encryption, we can easily retrieve the decrypted password and use in our application.
+
+==== How to use Jasypt-SpringBoot
+For projects not using `@SpringBootApplication` or `@EnableAutoConfiguration`, we can use the jasypt-spring-boot dependency directly
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.github.ulisesbocchio</groupId>
+    <artifactId>jasypt-spring-boot</artifactId>
+    <version>2.0.0</version>
+</dependency>
+----
+
+Please check for latest version link:https://mvnrepository.com/artifact/com.github.ulisesbocchio/jasypt-spring-boot[here] .
+
+let's encrypt the text `Password@2` with secret key `password` and add it to the `encrypted.properties`.
+
+[source,properties]
+----
+encrypted.property=ENC(dQWokHUXXFe+OqXRZYWu22BpXoRZ0Drt)
+----
+
+New configuration class is required for `Jasypt-springboot` dependency and we need to add the annotation `@EncryptablePropertySource` at the class level.
+
+Below is an example,
+
+[source,java]
+----
+@Configuration
+@EncryptablePropertySource("encrypted.properties")
+public class AppJasyptConfig {
+}
+----
+
+And a service class to map the property in encrypted.properties.
+[source,java]
+----
+@Service
+public class PropertyMapperJasypt {
+
+    @Value("${encrypted.property}")
+    private String property;
+
+    public String getProperty() {
+        return property;
+    }
+}
+----
+Using the above service class and setting the secret key which we used for encryption, we can easily retrieve the encrypted.property.
```

Comment:
This is an example that sounds more like a "tutorial to get started with jasypt" to me. IMHO we need a clear pattern/best practice how to do it properly.